lms/djangoapps/courseware/views/views.py

```diff
@@ -1485,6 +1487,34 @@ def enclosing_sequence_for_gating_checks(block):
     return None
 
 
+def _check_sequence_exam_access(request, location):
+    """
+    Checks the client request for an exam access token for a sequence.
+    Exam access is always granted at the sequence block. This method of gating is
+    only used by the edx-exams system and NOT edx-proctoring.
+    """
+    if not exams_ida_enabled(location.course_key):
+        return True
+
+    if request.user.is_staff or is_masquerading_as_specific_student(request.user, location.course_key):
+        return True
+
+    exam_access_token = request.COOKIES.get(settings.EXAM_ACCESS_TOKEN_COOKIE)
```

Comment:
Some cookie questions:

1. How big are these tokens?
2. What domain is this cookie being written to (just the subdomain or the LMS, or some shared higher level domain)?
3. Do we have to worry about exceeding header limits on cookies from adding another JWT there?
4. Does this mean a user's browser is only ever authorized for one exam at a time? (That's probably fine–I'm just trying to make sure I understand how this is supposed to work.)
5. If I went to it with another browser (say I discover halfway through the exam that something's broken in Firefox and I log in again with Chrome), does edx-exams give me the cookie again?

Reply:
1. Seems to run 600ish characters, not sure how variable that becomes based on signing keys
2. Would have to be the next level shared between the lms and exams so `.edx.org`
3. I'll have to look into this
4. You should only be able to take one exam at a time
5. Yes the UI will be responsible for continually refreshing this cookie